unified-installer/astra-unified-installer.sh

```diff
@@ -99,15 +99,38 @@ readonly __FATAL=50
 #     captured_stdout="$(curl -sS https://bad-url.com 2> "$_ERR_FILE")"
 #     captured_stderr="$(get_captured_err)"
 readonly __ERR_FILE="tmp_last_captured_error.txt"
+
+# __TMP_ENV is used to store the user's env vars so that we can then re-apply them after having sourced
+# their config file. This allows us to make command line vars take precedence over what's in the config file.
+readonly __TMP_ENV="tmp.env"
+
 readonly __NEWLINE=$'\n' # This is for readability
 
 #----------------------------------------------------------------------
 #-- Script config
 #----------------------------------------------------------------------
+process_args() {
+    for arg in "$@"; do
+        case $arg in
+            --help|-h)
+                print_help
+                exit 0
+                ;;
+            --required)
+                local msg="Listed below are all required environment variables which do not have a default value"
+                msg+=" (and so must be provided by the user):${__NEWLINE}"
+                echo "$msg"
+                print_help | grep '\*[A-Z]\+'
+                exit 0
+                ;;
+        esac
+    done
+}
```

Comment:
This is just a typical arg processing function. Right now, there's only `--help/-h` and `--required`. About the latter flag -- I thought that, with the very long list of variables available, having a flag that outputs only the variables that are absolutely required (i.e. we don't provide a default for) might be useful. If you think it's fluff and not very useful though, I could be convinced.

Reply:
Why not put the required args in the help output? I think that's where a customer would expect to find it. Most people assume there is a -h or --help on CLI programs but would have to remember/read the docs to learn about the --required arg.

Reply:
"Listed below are all required environment variables which do not have a default value" maybe makes more sense to just say "Listed below are all required environment variables"

Reply:
The required args are marked with `*` in the help output, but that gets somewhat lost in the sea of variables available, which is why the `--required` exists. We could re-order the help output to list the required variables first, but I thought it would be better to keep everything categorized properly. Maybe not? 

As for the wording, yeah, yours is better. I got a bit lost in the technicality initially since a lot of variables that have default _are_ required for the script to work, while some work fine even if empty, but from a user's perspective, they only care about what's required for them to input, not the technical behind-the-scenes meaning of "required". I'll change the message for it.